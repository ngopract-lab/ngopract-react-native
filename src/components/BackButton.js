import React, { Component, PropTypes } from 'react';
import {
  View,
  Text,
  TouchableHighlight,
  Image
} from 'react-native';

import { styles } from '../routes/style';

export const BackButton = props => (
  <TouchableHighlight
    onPress={props.onBack}
    style={styles.backButton}
    underlayColor='transparent'>
    { props.title !== 'HOME' ?
      <View style={styles.backButtonContainer}>
        <Image
          source={require('../images/back.png')}
          style={{width: 15, height: 15}}/>
        <Text style={styles.backButtonText}>
          BACK
        </Text>
      </View> :
      <Text></Text> }

  </TouchableHighlight>
);
