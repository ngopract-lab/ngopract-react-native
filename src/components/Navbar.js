import React, { Component, PropTypes } from 'react';
import {
  View,
  Text,
  TouchableHighlight,
  ScrollView,
  Image
} from 'react-native';

import { styles } from '../routes/style';

export const Navbar = props => {
  console.log(props);
  return (
  <ScrollView style={styles.navbar} horizontal={true}>
    <TouchableHighlight
      style={styles.navbarTab}
      onPress={props.goToHome}
      underlayColor='transparent'>
      <Text style={styles.navbarTabText}>Home</Text>
    </TouchableHighlight>
    <View style={styles.navbarTab}>
      <Text style={styles.navbarTabText}>People</Text>
    </View>
    <View style={styles.navbarTab}>
      <Text style={styles.navbarTabText}>Messages</Text>
    </View>
    <View style={styles.navbarTab}>
      <Text style={styles.navbarTabText}>Profile</Text>
    </View>
  </ScrollView>
  );
};

Navbar.propTypes = {
  title: PropTypes.string.isRequired,
  onForward: PropTypes.func.isRequired,
  onBack: PropTypes.func.isRequired,
  goToHome: PropTypes.func.isRequired,
  goToProfile: PropTypes.func.isRequired,
}
