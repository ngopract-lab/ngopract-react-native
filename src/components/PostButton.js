import React, { Component, PropTypes } from 'react';
import {
  Text,
  TouchableHighlight,
} from 'react-native';

import { styles } from '../style';

export const PostButton = () => (
  <TouchableHighlight style={styles.postButton}>
    <Text style={styles.postButtonText}>+</Text>
  </TouchableHighlight>
);
