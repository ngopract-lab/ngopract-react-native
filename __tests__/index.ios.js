import 'react-native';
import React from 'react';
import Index from '../index.ios.js';

import { PostButton } from '../src/components/PostButton';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

// Basic Render Testing
it('Index renders correctly', () => {
  const tree = renderer.create(
    <Index />
  );
});

// Post Button Snapshot Testing
/* What is snapshot testing?
https://facebook.github.io/jest/docs/snapshot-testing.html#content
*/
test('PostButton renders correctly', () => {
  const tree = renderer.create(
    <PostButton />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
