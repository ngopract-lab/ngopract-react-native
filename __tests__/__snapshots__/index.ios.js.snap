exports[`test PostButton renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#2574A9",
        "borderRadius": 1000,
        "bottom": 20,
        "height": 80,
        "justifyContent": "center",
        "position": "absolute",
        "right": 20,
        "width": 80,
      },
    ]
  }
  testID={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFF",
        "fontSize": 50,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }>
    +
  </Text>
</View>
`;
